Pod/Classes/WPVideoPlayerView.m

```diff
@@ -41,18 +45,54 @@ - (void)commonInit {
     self.player = [[AVPlayer alloc] init];
     self.playerLayer = [AVPlayerLayer playerLayerWithPlayer: self.player];
     [self.layer addSublayer: self.playerLayer];
+    [self addSubview:self.controlToolbar];
+
+    __weak __typeof__(self) weakSelf = self;
+    self.timeObserver = [self.player addPeriodicTimeObserverForInterval:CMTimeMakeWithSeconds(1, NSEC_PER_SEC) queue:nil usingBlock:^(CMTime time) {
+        [weakSelf updateVideoDuration];
+    }];
 }
 
 - (void)dealloc {
-    [self.playerItem removeObserver:self forKeyPath: @"status"];
+    [_playerItem removeObserver:self forKeyPath: @"status"];
     [[NSNotificationCenter defaultCenter] removeObserver: self];
-    [self.player pause];
+    [_player removeTimeObserver:self.timeObserver];
+    [_player pause];
     _asset = nil;
+    _player = nil;
 }
 
 - (void)layoutSubviews {
     [super layoutSubviews];
     self.playerLayer.frame = self.bounds;
+    CGFloat position = toolbarHeight;
+    if (self.controlToolbarHidden) {
+        position = 0;
+    }
+    self.controlToolbar.frame = CGRectMake(0, self.frame.size.height - position, self.frame.size.width, toolbarHeight);
+}
+
+- (UIToolbar *)controlToolbar {
+    if (_controlToolbar) {
```

Comment:
Total nitpick, but this method and `videoDurationButton` handle an existing ivar differently – this one checks if it exists, `videoDurationButton` checks if it's nil. Can we do it the same way for both? :)